src/Polly.Core/Simmy/Outcomes/OutcomeChaosStrategy.cs

```diff
@@ -0,0 +1,132 @@
+
+    public Func<OutcomeGeneratorArguments, ValueTask<Outcome<T>?>>? OutcomeGenerator { get; }
+
+    public Func<OutcomeGeneratorArguments, ValueTask<Outcome<Exception>?>>? FaultGenerator { get; }
+
+    public Outcome<T>? Outcome { get; private set; }
+
+    public Outcome<Exception>? Fault { get; }
+
+    protected internal override async ValueTask<Outcome<T>> ExecuteCore<TState>(Func<ResilienceContext, TState, ValueTask<Outcome<T>>> callback, ResilienceContext context, TState state)
+    {
+        try
+        {
+            if (await ShouldInjectAsync(context).ConfigureAwait(context.ContinueOnCapturedContext))
+            {
+                if (FaultGenerator is not null)
+                {
+                    var fault = await InjectFault(context).ConfigureAwait(context.ContinueOnCapturedContext);
+                    if (fault is not null)
+                    {
+                        return new Outcome<T>(fault);
+                    }
+                }
+                else if (OutcomeGenerator is not null)
+                {
+                    var outcome = await InjectOutcome(context).ConfigureAwait(context.ContinueOnCapturedContext);
+                    if (outcome.HasValue)
+                    {
+                        return new Outcome<T>(outcome.Value.Result);
+                    }
+                }
+                else
+                {
+                    throw new InvalidOperationException("Either a fault or fake outcome to inject must be defined.");
+                }
+            }
+
+            return await StrategyHelper.ExecuteCallbackSafeAsync(callback, context, state).ConfigureAwait(context.ContinueOnCapturedContext);
+        }
+        catch (OperationCanceledException e)
+        {
+            return new Outcome<T>(e);
+        }
+    }
+
+    private async ValueTask<Outcome<T>?> InjectOutcome(ResilienceContext context)
+    {
+        var outcome = await OutcomeGenerator!(new(context)).ConfigureAwait(context.ContinueOnCapturedContext);
+        var args = new OutcomeArguments<T, OnOutcomeInjectedArguments>(context, outcome.Value, new(context));
+        _telemetry.Report(new(ResilienceEventSeverity.Warning, OutcomeConstants.OnOutcomeInjectedEvent), context, args);
+
+        if (OnOutcomeInjected is not null)
+        {
+            await OnOutcomeInjected(args).ConfigureAwait(context.ContinueOnCapturedContext);
+        }
+
+        return outcome;
+    }
+
+    private async ValueTask<Exception?> InjectFault(ResilienceContext context)
+    {
+        try
+        {
+            var fault = await FaultGenerator!(new(context)).ConfigureAwait(context.ContinueOnCapturedContext);
+            if (!fault.HasValue)
+            {
+                return null;
+            }
+
+            // to prevent injecting the fault if it was cancelled while executing the FaultGenerator
+            context.CancellationToken.ThrowIfCancellationRequested();
+
+            Outcome = new(fault.Value.Exception!);
```

Comment:
Is this correct? Why override the default?

Reply:
I thought it would make it more consistent overriding the outcome with the current fault that was injected, but that's a good point, overriding it does not feel right, I'll remove it.